SUL23-60 Style the header and footer components

pookmish opened the pull request:
# READY FOR REVIEW

# Summary
- Styles and interaction for the main menu
- Styles for the secondary navigation on the left sidebar.

# Steps to Test

1. view the preview link from netlify
2. interact with the main menu in mobile and on desktop
3. verify you can access everything and it function as you would expect it to.



<a href="https://gitpod.io/#https://github.com/SU-SWS/sulgryphon-nextjs/pull/30"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

jdwjdwjdw commented on components/patterns/lockup.tsx:
```diff
@@ -1,19 +1,24 @@
+              className="lg:su-h-[45px] su-logo su-text-cardinal-red su-type-4 su-leading-[initial] su-pr-5 su-mr-5 lg:su-border-r su-border-solid su-border-black">
+              Stanford
+            </div>
+            <div className="su-text-20 lg:su-text-m2 su-font-regular su-text-black su-flex su-items-end">
```
Can you make this uppercase to match figma?

jdwjdwjdw commented on components/patterns/lockup.tsx:
```diff
@@ -1,19 +1,24 @@
+          <Logo/>
+          <div className="lg:su-flex">
+            <div
+              className="lg:su-h-[45px] su-logo su-text-cardinal-red su-type-4 su-leading-[initial] su-pr-5 su-mr-5 lg:su-border-r su-border-solid su-border-black">
```
Looks like the recent [decanter font addition](https://github.com/SU-SWS/sulgryphon-nextjs/pull/24/files#diff-4ee0e71a2145586038d583d58bcbd7aaa6b42318f417f5b62f2cbee48931861b) are causing some adjustment issues that will need to be fixed. And can we add some more spacing between the logo and `Stanford`?
<img width="240" alt="Screen Shot 2022-11-21 at 12 01 53 PM" src="https://user-images.githubusercontent.com/49299083/203147922-693aa036-c83d-49af-8c3c-16deb475077e.png">

jdwjdwjdw reviewed (changes requested):
Added some notes - looks great overall. Here a few additional questions:
- Why is the `Placeholder no link` dropdown higher than the others/have a font-weight of 700 vs 600 of the others?
<img width="650" alt="Screen Shot 2022-11-21 at 11 11 31 AM" src="https://user-images.githubusercontent.com/49299083/203153603-85e46536-94a7-4b1b-8484-ba778cd4269d.png">
- I'm Currently unable to see / access dropdown items beyond level 2 if it's the last menu item (getting cut off - )
<img width="285" alt="image" src="https://user-images.githubusercontent.com/49299083/203153981-37b28811-60ac-4992-8145-e48eb717e733.png">
<img width="356" alt="Screen Shot 2022-11-21 at 11 18 59 AM" src="https://user-images.githubusercontent.com/49299083/203153781-48be37f9-9e41-4628-b638-61577afb5ade.png">
- Are we not worrying about the search part right now?
<img width="248" alt="Screen Shot 2022-11-21 at 12 35 27 PM" src="https://user-images.githubusercontent.com/49299083/203154211-50916c66-3f70-410a-8668-4acad29131ec.png">
- Looks like figma is showing a redline on hover for dropdown icons, and sub-menu links
<img width="189" alt="image" src="https://user-images.githubusercontent.com/49299083/203154345-1ab85053-f9a7-45c9-8d8b-e6f4e1d5f7e8.png">
<img width="299" alt="image" src="https://user-images.githubusercontent.com/49299083/203154440-0f2033ad-07bf-409c-9047-66b8b3f963fe.png">

pookmish commented:
I've adjusted some of your feedback and also implemented some changes after review with Ishi. Gonna merge this as we'll continue some iterative improvements on the menu later.
